common/cpp/verify_dcap_report/verify-dcap-quote.cpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "verify-dcap-quote.h"
+#include "verify-dcap-quote.h"
+
+#include <sgx_dcap_quoteverify.h>
+#include "hex_string.h"
+#include "sgx_dcap_ql_wrapper.h"
+#include "sgx_quote_3.h"
+#include "sgx_report.h"
+#include "sgx_error.h"
+#include "sgx_qve_header.h"
+#include "sgx_ql_quote.h"
+#include "utils.h"
+
+bool verify_dcap_quote_signature(
+    const char* dcap_attestation_signing_cert_pem,
+    const char* dcap_report,
+    unsigned int dcap_report_len,
+    char* signature,
+    unsigned int signature_len) {
+    // TODO: Implement DCAP quote signature
+    return true;
+}
+
+bool verify_dcap_quote(const char* dcap_quote, int dcap_quote_len) {
+    int ret = 0;
+    time_t current_time = 0;
+    uint32_t supplemental_data_size = 0;
+    uint8_t *p_supplemental_data = NULL;
+    quote3_error_t dcap_ret = SGX_QL_ERROR_UNEXPECTED;
+    sgx_ql_qv_result_t quote_verification_result = SGX_QL_QV_RESULT_UNSPECIFIED;
+    uint32_t collateral_expiration_status = 1;
+
+    dcap_ret = sgx_qv_get_quote_supplemental_data_size(&supplemental_data_size);
+    if (dcap_ret == SGX_QL_SUCCESS && \
+        supplemental_data_size == sizeof(sgx_ql_qv_supplemental_t)) {
+        printf("\tInfo: sgx_qv_get_quote_supplemental_data_size successfully returned.\n");
```

Comment:
Instead of printf can we use logger functions if we have any.

Reply:
I shall replace printf with logger in another PR.